src/aset.rs

```diff
@@ -237,6 +237,11 @@ impl AutomaticSet {
     pub fn ensure_dfa(&mut self) {
         self.automaton.ensure_dfa();
     }
+
+    pub fn is_empty(self) -> bool {
+        let dfa = self.to_dfa().minimize();
```

Comment:
to_dfa() already minimizes automaton. The invariant is that Dfa is hold only in minimal form.